IdentityCore/src/util/NSDictionary+MSIDExtensions.m

```diff
@@ -96,6 +96,46 @@ - (NSDictionary *)dictionaryByRemovingFields:(NSArray *)fieldsToRemove
     return mutableDict;
 }
 
+- (BOOL)msidAssertTypeIsOneOf:(NSArray<Class> *)types ofKey:(NSString *)key required:(BOOL)required error:(NSError **)error
```

Comment:
didn't we have something similar already?

Reply:
Refactored old code to use this method.